internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -87,6 +87,7 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		}
 		log.Error(err, "Failed to retrieve Node by ProviderID")
 		r.recorder.Event(machine, corev1.EventTypeWarning, "Failed to retrieve Node by ProviderID", err.Error())
+		conditions.MarkUnknown(machine, clusterv1.MachineNodeHealthyCondition, clusterv1.NodeConditionsFailedReason, "Failed to get the node which is hosting this component: %v", err)
```

Comment:
The reason doesn't seem to match the error:
```go
	// NodeConditionsFailedReason (Severity=Warning) documents a node is not in a healthy state due to the failed state of at least 1 Kubelet condition.
	NodeConditionsFailedReason = "NodeConditionsFailed"
```

Reply:
Referring `PodInspectionFailedReason` [case](https://github.com/kubernetes-sigs/cluster-api/blob/5c38bf731e20be057d4e92f2d4ebbc89d966a31d/controlplane/kubeadm/api/v1beta1/condition_consts.go#L109), add new constant `NodeInspectionFailedReason` because it is very similar case like that.

d329648e

Reply:
Sounds good!